src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/RetryPolicyUtils.cs

```diff
@@ -357,6 +377,38 @@ public static void RaiseSchemaAmbientRetryMessage(RetryState retryState, int err
             }
         }
 
+        /// <summary>
+        /// Wait for SqlConnection to handle sleeping serverless instances (allows for them to wake up).
+        /// </summary>
+        public static SqlRetryLogicBaseProvider ServerlessWaitRetryLogicProvider()
+        {
+            var serverlessRetryLogic = new SqlRetryLogicOption
+            {
+                NumberOfTries = _serverlessMaxRetries,
+                MaxTimeInterval = TimeSpan.FromSeconds(_serverlessMaxIntervalTime),
+                DeltaTime = TimeSpan.FromSeconds(1),
+                TransientErrors = _retryableExtendedNetworkConnectivityError
+            };
+
+            var provider = SqlConfigurableRetryFactory.CreateFixedRetryProvider(serverlessRetryLogic);
+
+            provider.Retrying += (object s, SqlRetryingEventArgs e) =>
+            {
+                int attempts = e.RetryCount + 1;
+                Logger.Write(TraceEventType.Information, $"attempt {attempts} - current delay time:{e.Delay} \n");
```

Comment:
Perhaps use System.Environment.NewLine which does "the right thing" in a more portable way?

Reply:
Why do we even need a newline? Generally log messages would each be on their own new line anyways (or if not, a version that adds it such as Console.Write vs Console.WriteLine would be available)

Reply:
This was modified from some code (Example for retry provider), so I'm going to fix this based on feedback